tests/routes/PlayerInput.spec.ts

```diff
@@ -35,18 +37,21 @@ describe('PlayerInput', function() {
     player.process([['1'], ['Power Plant:SP']]);
     const options = player.getWaitingFor() as OrOptions;
     options.options.push(new UndoActionOption());
-    scaffolding.ctx.gameLoader.restoreGameAt = function(_gameId: string, _lastSaveId: number, cb: (game: Game | undefined) => void) {
-      cb(undo);
-    };
-    scaffolding.post(PlayerInput.INSTANCE, res);
-    scaffolding.req.emit('data', JSON.stringify([[String(options.options.length - 1)], ['']]));
-    scaffolding.req.emit('end');
+    scaffolding.ctx.gameLoader.restoreGameAt = (_gameId: string, _lastSaveId: number) => Promise.resolve(undo);
+    const post = scaffolding.asyncPost(PlayerInput.INSTANCE, res);
+    req.emit('data', JSON.stringify([[String(options.options.length - 1)], ['']]));
```

Comment:
This emission is happening too early before we are listening. This is happening because `Promise.resolve()` ands up running on the next _tick_. 

This snippet may help show the point:

```
Promise.resolve().then(() => console.log("B"));
console.log("A");
```

This outputs "A" and then "B" since `Promise.resolve()` doesn't run until the next tick.

With this in mind I suggest the following solution.

```suggestion
const emit = Promise.resolve().then(() => {
    req.emit('data', JSON.stringify([[String(options.options.length - 1)], ['']]));
    req.emit('end');
});

try {
   await Promise.all([emit, post]);
}
```

Reply:
Thank you!